src/components/map/map-control.js

```diff
@@ -162,12 +187,109 @@ const MapLegendPanel = ({layers, isActive, scale, toggleMenuPanel, isExport}) =>
     <MapControlPanel
       scale={scale}
       header={'Layer Legend'}
-      onClick={toggleMenuPanel}
+      onClick={onToggleMenuPanel}
       isExport={isExport}
     >
       <MapLegend layers={layers}/>
     </MapControlPanel>
+);
+
+MapLegendPanel.displayName = 'MapControlPanel';
+
+const SplitMapButton = React.memo(({isSplit, mapIndex, onToggleSplitMap}) => (
+  <MapControlButton
+    active={isSplit}
+    onClick={e => {
+      e.preventDefault();
+      onToggleSplitMap(isSplit ? mapIndex : undefined);
+    }}
+    key={`split-${isSplit}`}
+    className="map-control-button split-map"
+    data-tip
+    data-for="action-toggle"
+  >
+    {isSplit ? <Delete height="18px" /> : <Split height="18px" />}
+    <MapControlTooltip
+      id="action-toggle"
+      message={
+        isSplit ? 'Close current panel' : 'Switch to dual map view'
+      }
+    />
+  </MapControlButton>
+));
+
+SplitMapButton.displayName = 'SplitMapButton';
+
+const Toggle3dButton = React.memo(({dragRotate, onTogglePerspective}) => (
+  <MapControlButton
+    onClick={e => {
+      e.preventDefault();
+      onTogglePerspective();
+    }}
+    active={dragRotate}
+    data-tip
+    data-for="action-3d"
+  >
+    <Cube3d height="22px" />
+    <MapControlTooltip
+      id="action-3d"
+      message={dragRotate ? 'Disable 3D Map' : '3D Map'}
+    />
+  </MapControlButton>
+));
+
+Toggle3dButton.displayName = 'Toggle3dButton';
+
+const StyledToolBar = styled(Toolbar)`
+  position: absolute;
+  right: 32px;
+`;
+
+const MapDrawPanel = React.memo(({editor, isActive, onToggleMenuPanel, onSetEditorMode}) => {
+  return (
+    <div style={{position: 'relative'}}>
+      {isActive ? (
+        <StyledToolBar show={isActive} direction="column">
+          <ToolbarItem
+            onClick={() => onSetEditorMode(EDITOR_MODES.EDIT_VERTEX)}
+            label="select"
+            icon={(<CursorClick height="22px"/>)}
+            active={editor.mode === EDITOR_MODES.EDIT_VERTEX}
+          />
+          <ToolbarItem
+            onClick={() => onSetEditorMode(EDITOR_MODES.DRAW_POLYGON)}
+            label="polygon"
+            icon={(<Polygon height="22px"/>)}
+            active={editor.mode === EDITOR_MODES.DRAW_POLYGON}
+          />
+          <ToolbarItem
+            onClick={() => onSetEditorMode(EDITOR_MODES.DRAW_RECTANGLE)}
+            label="rectangle"
+            icon={(<Rectangle height="22px"/>)}
```

Comment:
A better way is to pass in `Rectangle` and render it in `<ToolbarItem><props.icon height="22px"></ToolbarItem>`

Reply:
how would we handle all toolbarItems, each has a different icon.

Should we pass a default prop items with a list of button objects? something like this:

```
{
  onClick: () => onSetEditorMode(EDITOR_MODES.DRAW_POLYGON),
  label:'polygon',
  icon: Rectangle,
  activeCondition: EDITOR_MODES.DRAW_RECTANGLE
}
```